api/v1/controllers/samples.js

```diff
@@ -197,7 +201,10 @@ module.exports = {
    *  bulk upsert request has been received.
    */
   bulkUpsertSample(req, res/* , next */) {
-    utils.noReadOnlyFieldsInReq(req, helper);
+
+    // make the name post-able
+    const readOnlyFields = helper.readOnlyFields.filter((field) => field !== 'name');
+    utils.noReadOnlyFieldsInReq(req, readOnlyFields);
     const resultObj = { reqStartTime: new Date() };
```

Comment:
I am confused: are we going to remove line 207 in this pull request? If so, we'll need to add checks to both 
        redisModelSample.bulkUpsertSample(value, userName)  and 
        helper.model.bulkUpsertByName(value, userName).

Even if we remove line 207, we still need to iterate over the bulk upsert array somewhere in the code, correct? 
Where were we moving the noReadOnlyFieldsInReq to?